src/workflow/concurrency_manager.rs

```diff
@@ -51,7 +52,7 @@ impl WorkflowConcurrencyManager {
         } else {
             // Create a new workflow machines instance for this workflow, initialize it, and
             // track it.
-            let mut wfm = WorkflowManager::new(history, workflow_execution);
+            let mut wfm = WorkflowManager::new(history, workflow_execution, task_queue);
```

Comment:
You can avoid asyncing everything if you just make the value in the machines map a struct with the mutex and also the queue name. No need to pass it into the manager itself. Then you don't need to use `access` and fetching the queue name does not require async.